### PR TITLE
Adds melee aiming

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
Adds melee aiming. meant to go with https://github.com/BeeStation/BeeStation-Hornet/pull/2430

while on Walk and Harm intent, your melee attacks will hit mobs that are outside of softcrit if they are in the targeted tile, regardless of if you click their sprite or not

## Why It's Good For The Game
speedcombat is still cancer, this accompanies my earlier goal of slowing it down a bit

## Changelog
:cl:
add: you can now hit mobs in melee easier if you are on harm intent and walking (you only need to click the tile, not the sprite itself)
/:cl:

### CODE REVIEW

**Comment by TeomanTheGreat** [comment]
Changelog moment

**Comment by ike709** [comment]
Seems like an interesting idea.

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -41,6 +41,15 @@
 	return FALSE
 
 /obj/attackby(obj/item/I, mob/living/user, params)
```
Comment (ike709) [inline code/_onclick/item_attack.dm:43]: what

File: code/_onclick/item_attack.dm [hunk 2]
```diff
@@ -41,6 +41,15 @@
...
+	var/turf/T = (get_turf(src))
+	if(user.a_intent == INTENT_HARM && user.m_intent == MOVE_INTENT_WALK)//if you swing carefully without blocking, you click easier
+		for(var/mob/living/L in T)
```
Comment (ike709) [inline code/_onclick/item_attack.dm]:
```suggestion
		for(var/mob/living/L in get_turf(src))
```

Remove the above get_turf(), this reduces a lot of redundant calls.

File: code/_onclick/item_attack.dm [hunk 3]
```diff
@@ -41,6 +41,15 @@
...
+			if(L.stat)
```
Comment (ike709) [inline code/_onclick/item_attack.dm:46]: You sure about this?

File: code/game/objects/objs.dm [hunk 4]
```diff
@@ -334,6 +334,21 @@
 /obj/proc/plunger_act(obj/item/plunger/P, mob/living/user, reinforced)
 	return
 
+/obj/attack_hand(mob/user)
+	var/turf/T = (get_turf(src))
+	if(user.a_intent == INTENT_HARM && user.m_intent == MOVE_INTENT_WALK)//if you swing carefully without blocking, you click easier
+		for(var/mob/living/L in T)
```
Comment (ike709) [inline code/game/objects/objs.dm]: See other comments.

**Comment by RafRoq** [comment]
So now the meta will be walking combat ?

File: code/_onclick/item_attack.dm (hunk 3 above)
Comment (zeskorion) [reply code/_onclick/item_attack.dm:46]: Yeah. There's no need for aiming if they cant dodge, and you want to hit the guy with a gun, not the softcrit dude he's standing on

File: code/game/objects/objs.dm (hunk 4 above)
Comment (zeskorion) [reply code/game/objects/objs.dm]: Will do when j get home

**Comment by ike709** [comment]
@uBadush I mean your opponent can just run away so... sure?

File: code/_onclick/item_attack.dm [hunk 5]
```diff
@@ -41,6 +41,15 @@
 	return FALSE
 
 /obj/attackby(obj/item/I, mob/living/user, params)
+	var/turf/T = (get_turf(src))
```
Comment (ike709) [inline code/_onclick/item_attack.dm]: You still need to remove this line you chumbis.

Comment (zeskorion) [reply code/_onclick/item_attack.dm]: Im not at home and i cant webedit it. Ill do it when i get home 

As well as some shit with altclickon so holding alt to walk works

**Comment by zeskorion** [comment]
will attempt to change alt key to a toggle (as well as a refactor of things that use the walk intent to a proc that checks movement speed regardless of walking or not) in a future pr

**Comment by Archanial** [comment]
Doesn't it remove aiming?
Super speed is cancer, just slow everyone down. It's not like people won't get used to it.

**Comment by zeskorion** [comment]
> Doesn't it remove aiming?
> Super speed is cancer, just slow everyone down. It's not like people won't get used to it.

whaddya mean remove aiming? this only affects melee attacks, and only slightly

**Comment by Archanial** [comment]
> whaddya mean remove aiming? this only affects melee attacks, and only slightly

I mean you no longer have to aim at the sprite.
idk this could be hit or miss

**Comment by ivanmixo** [comment]
this seems like a crutch for being inaccurate with your clicks lmao

File: code/game/objects/objs.dm [hunk 6]
```diff
@@ -334,6 +334,20 @@
...
+			if(L.stat)
+				continue
+			if(user == L)
+				continue
+			L.attack_hand(user)
+			user.changeNext_move(CLICK_CD_MELEE)
+			return
+	. = ..()
+	if(.)
+		return
```
Comment (PowerfulBacon) [inline code/game/objects/objs.dm:349]: What's the point of this return, it's at the end of the proc anyway

**Comment by Isy232** [comment]
@zeskorion You can walk while being on a secway or scooter and still move at the same speed as running. This sounds kinda unbalanced.

**Comment by zeskorion** [comment]
> @zeskorion You can walk while being on a secway or scooter and still move at the same speed as running. This sounds kinda unbalanced.

That will be fixed at a later date

**Comment by ike709** [comment]
> That will be fixed at a later date

A later date mayhaps, but still prior to merge. Scooters shall not become the new meta.

**Comment by zeskorion** [comment]
> > That will be fixed at a later date
> 
> A later date mayhaps, but still prior to merge. Scooters shall not become the new meta.

Nah the plan was to refactor all the shit thar chexks walk to a special chec_careful_movement() proc, which checks current speed mods instead of if you're actually using the Walk intent, as well as other factors

**Comment by eLeCtrOssSnake** [comment]
Kinda seems like shit tbh.

**Comment by zeskorion** [comment]
(waiting on another pr)

**Comment by utilbird** [comment]
It’ll make pvp a bit easier for those with higher ping 👍

**Comment by utilbird** [comment]
Shame 😔

**Comment by zeskorion** [comment]
Why mark this as stale? Its waiting on another merge

**Comment by ike709** [comment]
You still need to address scooter meta before this is getting in.

**Comment by ZombiesVsPlants** [comment]
huh, have to be on harm intent to aim, and another to block?

**Comment by ike709** [comment]
Keeping this stale unless you address the new meta this introduces.

**Comment by utilbird** [comment]
image

**Comment by zeskorion** [comment]
> Keeping this stale unless you address the new meta this introduces.

it was addressed in another pr but i got bored of coding so i dont care

**Comment by zeskorion** [comment]
mk, 10 years later...
https://github.com/BeeStation/BeeStation-Hornet/pull/2583
this is the pr this is dependent on, both need to be reopenned.

**Pull request closed without merging**